Firestore/Source/Core/FSTQuery.h

```diff
@@ -38,15 +38,27 @@ typedef NS_ENUM(NSInteger, FSTRelationFilterOperator) {
 };
 
 /** Interface used for all query filters. */
-@protocol FSTFilter <NSObject>
+@interface FSTFilter : NSObject
 
-/** Returns the field the Filter operates over. */
+/**
+ * Creates a filter for the provided path, operator, and value.
+ *
+ * Note that if the relational operator is FSTRelationFilterOperatorEqual and
+ * the value is [FSTNullValue nullValue] or [FSTDoubleValue nanValue], this
+ * will return the appropriate FSTNullFilter or FSTNanFilter class instead of a
+ * FSTRelationFilter.
+ */
++ (instancetype)createWithField:(const firebase::firestore::model::FieldPath &)field
```

Comment:
Objective-C functions that create a thing name it as a first part of the label. There's usually no explicit "create" or "make" (e.g. `NSString stringWithFormat:`). This should therefore start with "filter" (as does FSTRelationFilter's factory).

The labels here should match the ones in `FSTRelationFilter`. This is `...WithField:op:value`, that one is `...WithField:filterOperator:value`. Objective-C style favors plain language over abbreviation, so probably filterOperator is better. (This used to be just "operator" but that works in Objective-C but not Objective-C++ where "operator" is a reserved word. This is one of the few areas where Objective-C++ is not a strict superset of Objective-C.)

Reply:
Done.

Fun fact: Actually using this method (eg api/firquery.mm) uses the word "filter" 6 times in a single line.
```
FSTFilter *filter = [FSTFilter filterWithField:fieldPath filterOperator:filterOperator value:field];
```

Reply:
It wouldn't be Objective-C if you weren't repeating yourself at least three times.